stl/inc/memory

```diff
@@ -1251,11 +1252,62 @@ private:
         _Owner._Call_deleter = false;
     }
 
+    // clang-format off
     template <class _Ty0, class... _Types>
+#ifdef __cpp_concepts
+        requires (!is_array_v<_Ty0>)
+#endif // __cpp_concepts
     friend shared_ptr<_Ty0> make_shared(_Types&&... _Args);
+    // clang-format on
 
+#ifdef __cpp_concepts
+    // clang-format off
+    template <class _Ty0>
+        requires is_unbounded_array_v<_Ty0>
+    friend shared_ptr<_Ty0> make_shared(const size_t _Count);
+
+    template <class _Ty0>
+        requires is_unbounded_array_v<_Ty0>
+    friend shared_ptr<_Ty0> make_shared(const size_t _Count, const remove_extent_t<_Ty0>& _Val);
+
+    template <class _Ty0>
+        requires is_bounded_array_v<_Ty0>
+    friend shared_ptr<_Ty0> make_shared();
+
+    template <class _Ty0>
+        requires is_bounded_array_v<_Ty0>
+    friend shared_ptr<_Ty0> make_shared(const remove_extent_t<_Ty0>& _Val);
+    // clang-format on
+#endif // __cpp_concepts
+
+    // clang-format off
     template <class _Ty0, class _Alloc, class... _Types>
+#ifdef __cpp_concepts
+        requires (!is_array_v<_Ty0>)
+#endif // __cpp_concepts
     friend shared_ptr<_Ty0> allocate_shared(const _Alloc& _Al_arg, _Types&&... _Args);
+    // clang-format on
+
+#ifdef __cpp_concepts
+    // clang-format off
+    template <class _Ty0, class _Alloc>
+        requires is_unbounded_array_v<_Ty0>
+    friend shared_ptr<_Ty0> allocate_shared(const _Alloc& _Al_arg, const size_t _Count);
+
+    template <class _Ty0, class _Alloc>
+        requires is_unbounded_array_v<_Ty0>
+    friend shared_ptr<_Ty0> allocate_shared(
+        const _Alloc& _Al_arg, const size_t _Count, const remove_extent_t<_Ty0>& _Val);
+
+    template <class _Ty0, class _Alloc>
+        requires is_bounded_array_v<_Ty0>
+    friend shared_ptr<_Ty0> allocate_shared(const _Alloc& _Al_arg);
+
+    template <class _Ty0, class _Alloc>
+        requires is_bounded_array_v<_Ty0>
+    friend shared_ptr<_Ty0> allocate_shared(const _Alloc& _Al_arg, const remove_extent_t<_Ty0>& _Val);
+    // clang-format on
+#endif // __cpp_concepts

```

Comment:
I belive it would be cleaner if you would split it into a `__cpp_concepts` section and an `#else` block

```suggestion
#ifdef __cpp_concepts
    // clang-format off
    template <class _Ty0, class... _Types>
        requires (!is_array_v<_Ty0>)
    friend shared_ptr<_Ty0> make_shared(_Types&&... _Args);

    template <class _Ty0>
        requires is_unbounded_array_v<_Ty0>
    friend shared_ptr<_Ty0> make_shared(const size_t _Count);

    template <class _Ty0>
        requires is_unbounded_array_v<_Ty0>
    friend shared_ptr<_Ty0> make_shared(const size_t _Count, const remove_extent_t<_Ty0>& _Val);

    template <class _Ty0>
        requires is_bounded_array_v<_Ty0>
    friend shared_ptr<_Ty0> make_shared();

    template <class _Ty0>
        requires is_bounded_array_v<_Ty0>
    friend shared_ptr<_Ty0> make_shared(const remove_extent_t<_Ty0>& _Val);

    template <class _Ty0, class _Alloc, class... _Types>
        requires (!is_array_v<_Ty0>)
    friend shared_ptr<_Ty0> allocate_shared(const _Alloc& _Al_arg, _Types&&... _Args);
    
    template <class _Ty0, class _Alloc>
        requires is_unbounded_array_v<_Ty0>
    friend shared_ptr<_Ty0> allocate_shared(const _Alloc& _Al_arg, const size_t _Count);

    template <class _Ty0, class _Alloc>
        requires is_unbounded_array_v<_Ty0>
    friend shared_ptr<_Ty0> allocate_shared(
        const _Alloc& _Al_arg, const size_t _Count, const remove_extent_t<_Ty0>& _Val);

    template <class _Ty0, class _Alloc>
        requires is_bounded_array_v<_Ty0>
    friend shared_ptr<_Ty0> allocate_shared(const _Alloc& _Al_arg);

    template <class _Ty0, class _Alloc>
        requires is_bounded_array_v<_Ty0>
    friend shared_ptr<_Ty0> allocate_shared(const _Alloc& _Al_arg, const remove_extent_t<_Ty0>& _Val);
    // clang-format on
#else // ^^^ __cpp_concepts / !__cpp_concepts vvv    
    template <class _Ty0, class... _Types>
    friend shared_ptr<_Ty0> make_shared(_Types&&... _Args);

    template <class _Ty0, class _Alloc, class... _Types>
    friend shared_ptr<_Ty0> allocate_shared(const _Alloc& _Al_arg, _Types&&... _Args);
#endif // !__cpp_concepts
```

Reply:
You're right, it definitely looks cleaner.